src/state/token_index.ml

```diff
@@ -0,0 +1,83 @@
+open Yojson
+open Lwt
+open Learnocaml_data
+
+
+let token_file = "sync/token.json"
+
+(* Unlocked *)
+let mutex_token = Lwt_mutex.create ()
+
+let cast_list l = `List l
+
+let string_to_json (value:string) = (`String value : Yojson.Basic.t)
+
+let token_to_string l = List.map (fun t -> Token.to_string t) l
+
+let string_to_token l = List.map (fun t -> Token.parse t) l
```

Comment:
```suggestion
let string_to_token l = List.map Token.parse l
```
the eta-expansion is unneeded here, but maybe this `string_to_token` function should be removed and inlined as well?